node/node.go

```diff
@@ -371,6 +371,25 @@ func (n *Node) GetESDTData(address, tokenID string, nonce uint64) (*esdt.ESDigit
 		return nil, err
 	}
 
+	if core.ESDTType(esdtToken.Type) != core.Fungible {
```

Comment:
you could have extracted this in a function like `loadESDTDataFromSystemAccount` and contained there all the if's with early exits

Reply:
this is not the check you need to do. If nonce > 0 it has metadata.

Reply:
used esdt storage handler